substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/macho/MachOObjectFile.java

```diff
@@ -80,7 +82,7 @@
      * Create an empty Mach-O object file.
      */
     public MachOObjectFile(int pageSize) {
-        this(pageSize, MachOCpuType.from(System.getProperty("svm.targetArch") == null ? System.getProperty("os.arch") : System.getProperty("svm.targetArch")));
+        this(pageSize, MachOCpuType.from(System.getProperty("svm.targetArch") == null ? ImageSingletons.lookup(Platform.class).getArchitecture() : System.getProperty("svm.targetArch")));
```

Comment:
How is it ever possible that `ImageSingletons.lookup(Platform.class).getArchitecture()` does not return the correct architecture? 

I think a goal of this PR should be to remove the system properties `svm.targetArch` and `svm.targetName`

Reply:
I was trying to make this PR additive without breaking any of existing use-cases (for example Gluon Substrate, or non public platforms). My reasoning was that after this PR is accepted, removal of these non-api ways to specify non standard `os.name` shouldn't be an issue.

Reply:
I'm talking here about internal usages of these system properties. I'm fine if we keep them set in the image builder for another release. But all internal usages should go away immediately in this PR.

Reply:
Oh now I get it, you are right. I'll change this asap.

Reply:
Should be fixed now.